src/Akka.Streams.Kafka/Stages/ProducerStage.cs

```diff
@@ -97,7 +105,21 @@ private void OnProducerError(object sender, Error error)
 
             if (!KafkaExtensions.IsBrokerErrorRetriable(error) && !KafkaExtensions.IsLocalErrorRetriable(error))
             {
-                FailStage(new Exception(error.Reason));
+                var exception = new KafkaException(error);
+                switch (_decider(exception))
+                {
+                    case Directive.Stop:
+                        // Throw
+                        _completionState.TrySetException(exception);
+                        FailStage(exception);
+                        break;
+                    case Directive.Resume:
+                        // TODO: resume the stream
+                        break;
+                    case Directive.Restart:
+                        // TODO: restart the stream
```

Comment:
What should I do here? Azure implementation try to take the next element or complete the stage